app/database/MetricsDB.scala

```diff
@@ -59,35 +59,36 @@ class MetricsDB(implicit val db: Database) {
   def getDistinctNewspaperBooks: Either[ProductionMetricsError, Seq[Option[String]]] =
     await(db.run(metricsTable.filter(book => !book.newspaperBook.isEmpty && book.newspaperBook =!= "").map(_.newspaperBook).distinct.result))
 
-  def getGroupedWordCounts(implicit filters: Filters): Either[ProductionMetricsError, List[GroupedWordCount]] = {
+  def getArticlesGroupedByLengthBounds(groupByCommissionedLength: Boolean = false)(implicit filters: Filters): Either[ProductionMetricsError, List[ArticleLengthGroup]] = {
 
     val lowerBoundsToUpperBounds: Map[Int, Int] = Map(
       0 -> 349,
       350 -> 649,
       650 -> 899
     )
 
+    def assignLowerBound(length: Rep[Option[Int]]): Case.TypedCase[Int, Int] =
+      (Case
+        If(length between(0, lowerBoundsToUpperBounds.get(0).get)) Then 0
+        If(length between(350, lowerBoundsToUpperBounds.get(350).get)) Then 350
+        If(length between(650, lowerBoundsToUpperBounds.get(650).get)) Then 650
+        If(length >= 900) Then 900)
+
     awaitWithTransformation(db.run(metricsTable
       .filter(Filters.originFilters)
-      .map( metric =>
-        Case
-          If(metric.wordCount between(0, lowerBoundsToUpperBounds.get(0).get)) Then 0
-          If(metric.wordCount between(350, lowerBoundsToUpperBounds.get(350).get)) Then 350
-          If(metric.wordCount between(650, lowerBoundsToUpperBounds.get(650).get)) Then 650
-          If(metric.wordCount >= 900) Then 900
-      )
+      .map( metric => if(groupByCommissionedLength) assignLowerBound(metric.commissionedWordCount) else assignLowerBound(metric.wordCount))
```

Comment:
This line is very long!

Reply:
Using types as opposed to a boolean flag has shortened this line